api/src/main/java/com/epam/pipeline/manager/datastorage/providers/ProviderUtils.java

```diff
@@ -45,4 +47,26 @@ public static String normalizeBucketName(String name) {
     public static String withoutTrailingDelimiter(final String path) {
         return StringUtils.isNotBlank(path) && path.endsWith(DELIMITER) ? path.substring(0, path.length() - 1) : path;
     }
+
+    public static <T> Long getSizeByPath(final Iterable<T> items, final String requestPath,
+                                         final Function<T, Long> getSize, final Function<T, String> getName) {
+        final boolean rootOrFolder = isRootOrFolder(requestPath);
+
+        Long folderSize = 0L;
+        for (final T item : items) {
+            if (rootOrFolder) {
+                folderSize += getSize.apply(item);
```

Comment:
I'm not sure but how `size` may be returned for folder? I thought that approach was to list objects resursively, in this case we should have only objects (files) not folders.

Reply:
comment added